### PR TITLE
Added bsf clean Command to Remove Autogenerated Files

### PR DESCRIPTION
**Pull request by RohanMishra315**
#120 

 ### Description:
This PR introduces a new bsf clean command that simplifies the cleanup of autogenerated files and directories during development. This feature is especially useful when testing new features or iterating locally, where manual deletion of these files is often needed.

 ### Key Features:
 1. New bsf clean Command:

The bsf clean command deletes the following autogenerated files and directories:
bsf/
bsf-result/
bsf.hcl
bsf.lock

2. Reused Cleanup Logic:

Reuses the cleanup logic from cmd/init, ensuring consistency and reducing code duplication.

### CODE REVIEW

File: cmd/clean/clean.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/buildsafe/bsf/cmd/init"
+	"os"
+)
+
+func init() {
+	rootCmd.AddCommand(cleanCmd)
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:10]: This logic should go over here:
https://github.com/buildsafedev/bsf/blob/a2c73d1a07c03290df94270b30c687bac5491b41/cmd/cmd.go#L70

File: cmd/clean/clean.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+var cleanCmd = &cobra.Command{
+	Use:   "clean",
+	Short: "Clean autogenerated BSF files and directories",
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:15]:
```suggestion
	Short: "cleans the autogenerated bsf files and directories",
```

File: cmd/clean/clean.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+			fmt.Printf("Error during cleanup: %v\n", err)
+		} else {
+			fmt.Println("Autogenerated BSF files and directories cleaned successfully.")
+		}
+	},
+}
+
+// cleanUp deletes the autogenerated BSF files and directories.
+func cleanUp() error {
+	configs := []string{"bsf", "bsf-result", "bsf.hcl", "bsf.lock"}
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:33]: Will this give an error if `bsf-result` is not already present? Please check once

File: cmd/clean/clean.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+		if err != nil {
+			fmt.Printf("Error during cleanup: %v\n", err)
+		} else {
+			fmt.Println("Autogenerated BSF files and directories cleaned successfully.")
+		}
+	},
+}
+
+// cleanUp deletes the autogenerated BSF files and directories.
+func cleanUp() error {
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:32]: Please reuse the logic which is already present under `cmd/init`

File: cmd/clean/clean.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+var cleanCmd = &cobra.Command{
+	Use:   "clean",
+	Short: "Clean autogenerated BSF files and directories",
+	Run: func(cmd *cobra.Command, args []string) {
+		if os.Getenv("BSF_DEBUG_MODE") == "" {
+			fmt.Println("BSF_DEBUG_MODE is not enabled. Skipping clean.")
+			return
+		}
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:20]: We don't really need this as the command will only be initialized if `BSF_DEBUG_MODE` is enabled.

File: cmd/clean/clean.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+		err := cleanUp() // Call the cleanUp function
+		if err != nil {
+			fmt.Printf("Error during cleanup: %v\n", err)
+		} else {
+			fmt.Println("Autogenerated BSF files and directories cleaned successfully.")
+		}
```
Comment (rakshitgondwal) [inline cmd/clean/clean.go:27]:
```suggestion
		if err := cleanUp(); err != nil {
			fmt.Println(styles.ErrorStyle.Render("error during cleanup:", err.Error()))
			os.Exit(1)
		}
		fmt.Println(styles.SucessStyle.Render("autogenerated bsf files/dirs cleaned successfully"))
```

**Review by rakshitgondwal — CHANGES_REQUESTED** [review]
Thanks for the PR! Can you please sign your commits?

**Comment by RohanMishra315** [comment]
> Thanks for the PR! Can you please sign your commits?

Done , pls check it and anything to add let me know

**Review by rakshitgondwal — COMMENTED** [review]
> Done , pls check it and anything to add let me know

Thanks for making the changes @RohanMishra315. The commits are still not signed and some of the comments are still unaddressed, please fix those as well. Additionally, can you add yourself in the `./CONTRIBUTORS.md` file as well?

File: cmd/clean/clean.go (hunk 3 above)
Comment (rakshitgondwal) [reply cmd/clean/clean.go]: Did you check this? Was it giving an error?

**Comment by rakshitgondwal** [comment]
Hi @RohanMishra315, let me know if you need any help

**Comment by rakshitgondwal** [comment]
Hi @RohanMishra315, checking up on this again

**Comment by rakshitgondwal** [comment]
Fixed in https://github.com/buildsafedev/bsf/pull/128

**Pull request closed without merging**